base/utils/R/ensemble.R

```diff
@@ -191,96 +214,194 @@ write.ensemble.configs <- function(defaults, ensemble.samples, settings, model,
+                           runid = run.id[[i]], 
+                           start.time = restart$start.time,
+                           stop.time =restart$stop.time, 
+                           settings = settings,
+                           new.state = new.state[i, ], 
+                           new.params = new.params[[i]], 
+                           inputs =list(met=list(path=inputs$samples[[i]])), 
+                           RENAME = TRUE)
+      )
     }
-    dir.create(file.path(settings$rundir, run.id), recursive = TRUE)
-    dir.create(file.path(settings$modeloutdir, run.id), recursive = TRUE)
-    
-    # write run information to disk
-    cat("runtype     : ensemble\n",
-        "workflow id : ", workflow.id, "\n",
-        "ensemble id : ", ensemble.id, "\n",
-        "run         : ", counter, "/", settings$ensemble$size, "\n",
-        "run id      : ", run.id, "\n",
-        "pft names   : ", as.character(lapply(settings$pfts, function(x) x[['name']])), "\n",
-        "model       : ", model, "\n",
-        "model id    : ", settings$model$id, "\n",
-        "site        : ", settings$run$site$name, "\n",
-        "site  id    : ", settings$run$site$id, "\n",
-        "met data    : ", settings$run$site$met, "\n",
-        "start date  : ", settings$run$start.date, "\n",
-        "end date    : ", settings$run$end.date, "\n",
-        "hostname    : ", settings$host$name, "\n",
-        "rundir      : ", file.path(settings$host$rundir, run.id), "\n",
-        "outdir      : ", file.path(settings$host$outdir, run.id), "\n",
-        file = file.path(settings$rundir, run.id, "README.txt"))
-    
-    do.call(my.write.config, args = list(
-      defaults = defaults, 
-      trait.values = lapply(
-        ensemble.samples, function(x, n) { x[n, , drop=FALSE] }, n=counter
-      ), 
-      settings = settings, 
-      run.id = run.id)
-    )
-    cat(run.id, file = file.path(settings$rundir, "runs.txt"), sep = "\n", append = TRUE)
+    params<-new.params
+    return(invisible(list(runs = data.frame(id=run.id), ensemble.id = ensemble.id, samples=list(met=inputs)
+                          )
+                     ))
   }
-
-  return(invisible(list(runs = runs, ensemble.id = ensemble.id)))
+  
+  
+  
 } # write.ensemble.configs
+
+
+
+#' Function for reurning the met ensuble based on sampling method, parent or etc
+#'
+#' @param settings list of PEcAn settings
+#' @param method Method for sampling - For now looping or sampling with replacement is implemented
+#' @param parenids If met's ids are read from a parent what are the ids
+#' @param ... 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+input.ens.gen<-function(settings,input,method="sampling",parenids=NULL,...){
+  #-- reading the dots and exposing them to the inside of the function
+  samples<-list()
+  dots<-list(...)
+  if (length(dots) > 0 ) lapply(names(dots), function(name){assign(name, dots[[name]], pos=1 )})
+
+  if ( tolower(input)=="met" ){
```

Comment:
bit confused -- this function looks like it will only run if input == met, but I don't see anything in the code below that is met specific other than `settings$run$inputs$met`, which could easily be abstracted away, e.g. `settings$run$inputs[[input]]`

Reply:
That said, aren't parameters a special case (need to be sampled from one file and passed as a list, rather than sampling file paths or IDs). I don't recall the earlier code handling that special case, so either there needs to be that case in this function or in the code that calls it.

Reply:
I was actually trying to reserve the possibility that this function could handle other components like soil and vegetation later on with different variable types. For the met we have just a vector of paths but just like parameters for soils we may have a list of dataframes or a list of lists. That's why I didn't want to assume that the rest of components also would have similar variable type as met. I felt like this would give the next developers the peace of mind that their work would be independent of met and parameters.
Parameters are also handled in line 262-263 of ensemble.R

Reply:
Yes, I see that there's code that handles the case where parameters samples is null, but that comes after the loop over everything that's in `samplingspace` so I wanted to make sure that this function handled being called for the `parameter` case correctly. It's not obvious to me that it does -- if parameters are handled later why not have an explicit return right at the start when input == parameters (and add a comment to that effect in the loop around line 254.

I do think you can assume that all other inputs are handled the same way as met -- this is why I was pushing you to not have input specific functions. Soils, veg, etc will all exist as ensembles of files since write.configs.[MODEL] always expects inputs to be passed in as file paths.

Reply:
Cool,... I'll change the code accordingly then.